scripts/build/configure/termux_step_configure_haskell_build.sh

```diff
@@ -0,0 +1,146 @@
+	fi
+
+	# Avoid gnulib wrapping of functions when cross compiling. See
+	# http://wiki.osdev.org/Cross-Porting_Software#Gnulib
+	# https://gitlab.com/sortix/sortix/wikis/Gnulib
+	# https://github.com/termux/termux-packages/issues/76
+	AVOID_GNULIB=""
+	AVOID_GNULIB+=" ac_cv_func_nl_langinfo=yes"
+	AVOID_GNULIB+=" ac_cv_func_calloc_0_nonnull=yes"
+	AVOID_GNULIB+=" ac_cv_func_chown_works=yes"
+	AVOID_GNULIB+=" ac_cv_func_getgroups_works=yes"
+	AVOID_GNULIB+=" ac_cv_func_malloc_0_nonnull=yes"
+	AVOID_GNULIB+=" ac_cv_func_posix_spawn=no"
+	AVOID_GNULIB+=" ac_cv_func_posix_spawnp=no"
+	AVOID_GNULIB+=" ac_cv_func_realloc_0_nonnull=yes"
+	AVOID_GNULIB+=" am_cv_func_working_getline=yes"
+	AVOID_GNULIB+=" gl_cv_func_dup2_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_fcntl_f_dupfd_cloexec=yes"
+	AVOID_GNULIB+=" gl_cv_func_fcntl_f_dupfd_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_fnmatch_posix=yes"
+	AVOID_GNULIB+=" gl_cv_func_getcwd_abort_bug=no"
+	AVOID_GNULIB+=" gl_cv_func_getcwd_null=yes"
+	AVOID_GNULIB+=" gl_cv_func_getcwd_path_max=yes"
+	AVOID_GNULIB+=" gl_cv_func_getcwd_posix_signature=yes"
+	AVOID_GNULIB+=" gl_cv_func_gettimeofday_clobber=no"
+	AVOID_GNULIB+=" gl_cv_func_gettimeofday_posix_signature=yes"
+	AVOID_GNULIB+=" gl_cv_func_link_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_lstat_dereferences_slashed_symlink=yes"
+	AVOID_GNULIB+=" gl_cv_func_malloc_0_nonnull=yes"
+	AVOID_GNULIB+=" gl_cv_func_memchr_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_mkdir_trailing_dot_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_mkdir_trailing_slash_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_mkfifo_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_mknod_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_realpath_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_select_detects_ebadf=yes"
+	AVOID_GNULIB+=" gl_cv_func_snprintf_posix=yes"
+	AVOID_GNULIB+=" gl_cv_func_snprintf_retval_c99=yes"
+	AVOID_GNULIB+=" gl_cv_func_snprintf_truncation_c99=yes"
+	AVOID_GNULIB+=" gl_cv_func_stat_dir_slash=yes"
+	AVOID_GNULIB+=" gl_cv_func_stat_file_slash=yes"
+	AVOID_GNULIB+=" gl_cv_func_strerror_0_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_strtold_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_symlink_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_tzset_clobber=no"
+	AVOID_GNULIB+=" gl_cv_func_unlink_honors_slashes=yes"
+	AVOID_GNULIB+=" gl_cv_func_unlink_honors_slashes=yes"
+	AVOID_GNULIB+=" gl_cv_func_vsnprintf_posix=yes"
+	AVOID_GNULIB+=" gl_cv_func_vsnprintf_zerosize_c99=yes"
+	AVOID_GNULIB+=" gl_cv_func_wcrtomb_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_wcwidth_works=yes"
+	AVOID_GNULIB+=" gl_cv_func_working_getdelim=yes"
+	AVOID_GNULIB+=" gl_cv_func_working_mkstemp=yes"
+	AVOID_GNULIB+=" gl_cv_func_working_mktime=yes"
+	AVOID_GNULIB+=" gl_cv_func_working_strerror=yes"
+	AVOID_GNULIB+=" gl_cv_header_working_fcntl_h=yes"
+	AVOID_GNULIB+=" gl_cv_C_locale_sans_EILSEQ=yes"
+
+	LLVM_BACKEND=""
+	[ "${TERMUX_ARCH}" != "i686" ] && {
+		LLVM_BACKEND="-fllvm --ghc-option=-fllvm"
+	}
+	[ "$TERMUX_DEBUG_BUILD" = "true" ] && OPTIMISATION="-O0" || OPTIMISATION="-O"
```

Comment:
I think it would be more logical to export this in one of the setup scripts if possible

Reply:
In `termux_step_setup_toolchain` as `TERMUX_HASKELL_LLVM_BACKEND` (or maybe simply `LLVM_BACKEND`)..?

Reply:
`TERMUX_HASKELL_LLVM_BACKEND`, and perhaps `TERMUX_HASKELL_OPTIMISATION` in `termux_step_setup_toolchain` sounds good to me!